src-java/opentsdb-topology/opentsdb-storm-topology/src/main/java/org/openkilda/wfm/topology/opentsdb/models/Storage.java

```diff
@@ -50,9 +50,15 @@ public DatapointValue get(Datapoint datapoint) {
         return map.get(createKey(datapoint));
     }
 
+    /**
+     * Removes datapoints from the storage if they are older than now() - ttlInMillis.
+     */
     public void removeOutdated(long ttlInMillis) {
         long now = System.currentTimeMillis();
         map.entrySet().removeIf(entry -> now - entry.getValue().getTime() > ttlInMillis);
+        if (map.isEmpty()) {
```

Comment:
I think it's better to resize the map anyway, because if there is one element left, we keep a map with one element and some bigger capacity.
There is a `HashMap` constructor taking a map that creates a new map with capacity just enough to accommodate the existing elements. As an option, it could be done either if the map is empty or the size is small enough so that the copy is not too expensive.

Reply:
Good point, but it is vary rare base. Because to face with the case we must
1. Had a significant capacity before.
2. Expire most of datapoints, but not all of them
3. Be in this state for a long time. (in other case the last datapoint will expire too).

I can't imagine how we can put system in such case.